src/common/pool_map.c

```diff
@@ -799,18 +788,6 @@ pool_tree_sane(struct pool_domain *tree, uint32_t version)
 		struct pool_domain *prev = &tree[0];
 		int		    child_nr = 0;
 
-		if (parent != NULL &&
-		    parent->do_comp.co_type >= tree[0].do_comp.co_type) {
-			D_DEBUG(DB_MGMT,
-				"Type of parent domain %d(%s) should be "
-				"smaller than child domain %d(%s)\n",
-				parent->do_comp.co_type,
-				pool_domain_name(parent),
-				tree[0].do_comp.co_type,
-				pool_domain_name(&tree[0]));
-			return false;
-		}
```

Comment:
Could you remind me why this needs to be removed? Sorry for forgetting the conclusion from our previous discussions.

Reply:
Was it because we would end up failing the check with parent type = 10 and child type = 10 when there are more than one RACK levels? With current one RACK level restriction, do we still need to remove the check?

Reply:
Right, this was when I was trying to support multiple levels. I should re-add this back in.